### PR TITLE
Button: Default hover state color

### PR DESCRIPTION
**Pull request by simkessy**
Issue: #979 

BEFORE: 

image

image

AFTER: 

image

## TODO

- [x] Accessibility checkup
- [x] Update documentation
- [x] Add/update Tests
- [x] Pinterest Designer (for design changes)
- [x] [Format PR title](https://github.com/pinterest/gestalt/#releasing): `ComponentName: Description`.

### CODE REVIEW

File: packages/gestalt/src/Button.js [hunk 1]
```diff
@@ -62,7 +61,8 @@ export default function Button(props: Props): Element<'button'> {
   } = props;
   const { name: colorSchemeName } = useColorScheme();
   // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
-  const isDarkModeRed = colorSchemeName === 'darkMode' && color === 'red';
+  const DARK_MODE = colorSchemeName === 'darkMode';
```
Comment (jennyscript) [inline packages/gestalt/src/Button.js:64]: Caps/underscore doesn't go with the naming style of the rest of the codebase. Could you name this like `isDarkMode` or `inDarkMode` maybe?

File: packages/gestalt/src/Button.js [hunk 2]
```diff
@@ -124,21 +125,3 @@ export default function Button(props: Props): Element<'button'> {
   );
   /* eslint-enable react/button-has-type */
 }
-
-Button.propTypes = {
```
Comment (chrislloyd) [inline packages/gestalt/src/Button.js:128]: Please don't remove these as part of this PR.

File: packages/gestalt/src/Button.css [hunk 3]
```diff
@@ -62,6 +66,17 @@
   background-color: #dadada;
 }
 
+.darkModeGray:hover,
+.darkModeGray:focus{
+  background-color: var(--gestalt-colorGray50)
+
+}
+.darkModeGray:active {
+  background-color: var(--gestalt-colorGray100)
```
Comment (jennyscript) [inline packages/gestalt/src/Button.css:75]: These need to be added to the file as well or they won't get built for IE

File: packages/gestalt/src/Button.js [hunk 4]
```diff
@@ -76,6 +77,7 @@ export default function Button(props: Props): Element<'button'> {
     [styles.enabled]: !disabled,
     [styles.inline]: inline,
     [styles.block]: !inline,
+    [styles.darkModeGray]: isDarkMode && color === 'gray',
```
Comment (jennyscript) [inline packages/gestalt/src/Button.js:80]: This needs to be part of the same logic as `colorClass` so it gets handled correctly with the disabled/selected logic. You can see the issue in the docs where in dark mode gray selected has a hover color (incorrect) but red selected does not (correct)

File: packages/gestalt/src/Button.css [hunk 5]
```diff
@@ -1,3 +1,9 @@
+:root {
+  --gestalt-colorGray200: #767676;
```
Comment (jennyscript) [inline packages/gestalt/src/Button.css:2]: colorGray200 isn't used in this file

**Pull request merged**